sklearn/utils/__init__.py

```diff
@@ -553,3 +554,21 @@ def get_chunk_n_rows(row_bytes, max_n_rows=None,
                       (working_memory, np.ceil(row_bytes * 2 ** -20)))
         chunk_n_rows = 1
     return chunk_n_rows
+
+
+def is_scalar_nan(x):
+    """Tests if x is NaN
+
+    This function is meant to overcome the issue that np.isnan does not allow
+    non-numerical types as input, and that np.nan is not np.float('nan').
+
+    Parameters
+    ----------
+    x : any type
+
+    Returns
+    -------
+    boolean
```

Comment:
Please add a small example doctest such as at the end of the docstring:

```python
>>> is_scalar_nan(np.nan)
True
>>> is_scalar_nan(float("nan"))
True
>>> is_scalar_nan(None)
False
>>> is_scalar_nan("")
False
```

And maybe a more extensive unit test in in `sklearn/utils/tests/test_utils.py`:

```python
def test_is_scalar_nan():
    assert is_scalar_nan(float("nan"))
    assert is_scalar_nan(np.nan)
    assert is_scalar_nan(np.float("nan"))
    assert is_scalar_nan(np.float32("nan"))
    assert is_scalar_nan(np.float64("nan"))

    assert not is_scalar_nan(0)
    assert not is_scalar_nan(0.)
    assert not is_scalar_nan(None)
    assert not is_scalar_nan("")
    assert not is_scalar_nan("nan")
```